src/transforms/aws_ec2_metadata.rs

```diff
@@ -112,7 +112,18 @@ impl TransformConfig for Ec2Metadata {
     fn build(&self, cx: TransformContext) -> crate::Result<Box<dyn Transform>> {
         let (read, write) = evmap::new();
 
-        let keys = Keys::new(&self.namespace);
+        // Check if the namespace is set to `""` which should mean that we do
+        // not want a prefixed namespace.
+        let namespace = self.namespace.clone().and_then(|namespace| {
+            if namespace == "" {
```

Comment:
Is it worth trimming the input here to be extra defensive? i.e. should `"  "` also be treated as empty?

Reply:
I guess but at this point this just sounds like something our insert should handle?

Reply:
That is a fair point.